packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -79,64 +67,116 @@ export type TooltipProps = MergeRight<
-
-    const { refs, floatingStyles, context } = useFloating({
-      open: internalOpen,
-      onOpenChange: setIsOpen,
-      placement,
-      whileElementsMounted: autoUpdate,
-      middleware: [
-        offset((data) => {
-          // get pseudo element arrow size
-          const styles = getComputedStyle(data.elements.floating, '::before');
-          return parseFloat(styles.height);
-        }),
-        flip({
-          fallbackAxisSideDirection: 'start',
-        }),
-        shift(),
-        arrowPseudoElement,
-      ],
-    });
-
-    const { styles: animationStyles } = useTransitionStyles(context, {
-      initial: {
-        opacity: 0,
-      },
-    });
-
-    const { getReferenceProps, getFloatingProps } = useInteractions([
-      // Event listeners to change the open state
-      useHover(context, { move: false, delay }),
-      useFocus(context),
-      useDismiss(context),
-      useRole(context, { role: 'tooltip' }),
-    ]);
-
-    const mergedRef = useMergeRefs([ref, refs.setFloating]);
-
-    const childMergedRef = useMergeRefs([
-      (children as ReactElement & RefAttributes<HTMLElement>)
-        .ref as MutableRefObject<HTMLElement>,
-      refs.setReference,
-    ]);
+    const randomTooltipId = useId();
+
+    const [internalOpen, setInternalOpen] = useState(false);
+
+    const triggerRef = useRef<HTMLElement>(null);
+    const tooltipRef = useRef<HTMLDivElement>(null);
+    const mergedRefs = useMergeRefs([tooltipRef, ref]);
+
+    const controlledOpen = open ?? internalOpen;
+
+    // NOTE: This code is purely to add React controlled component ability to Popover API
+    useEffect(() => {
+      if (!tooltipRef.current || !triggerRef.current) return;
+
+      const trigger = triggerRef.current;
+      const tooltip = tooltipRef.current;
+
+      const handleMouseover = () => {
+        setInternalOpen(true);
+      };
+
+      const handleMouseout = (event: MouseEvent) => {
+        /* if we move mosue to tooltip, don't close */
+        const el = event.relatedTarget as Element | null;
+        if (el === tooltip) return;
+
+        setInternalOpen(false);
+      };
+
+      const handleFocusIn = () => {
+        setInternalOpen(true);
+      };
+
+      const handleFocusOut = () => {
+        setInternalOpen(false);
+      };
+
+      tooltip?.togglePopover?.(controlledOpen);
+      trigger.addEventListener('mouseover', handleMouseover);
+      /* We use document so user can move to tooltip contents */
+      document.addEventListener('mouseout', handleMouseout);
+      trigger.addEventListener('focus', handleFocusIn);
+      trigger.addEventListener('focusout', handleFocusOut);
+
+      return () => {
+        trigger.removeEventListener('mouseover', handleMouseover);
+        document.removeEventListener('mouseout', handleMouseout);
+        trigger.removeEventListener('focus', handleFocusIn);
+        trigger.removeEventListener('focusout', handleFocusOut);
```

Comment:
Skulle vi brukt `focusin` siden vi bruker `focusout`,
Og `mouseenter` i stede for `mouseover` siden den vil trigge skjeldnere dersom det er flere barn? ☺️